bridge/ApiRequestHandler.cpp

```diff
@@ -489,34 +500,48 @@ httpd::Response ApiRequestHandler::onRequest(const httpd::Request& request)
                 }
                 else
                 {
-                    logger::warn("failed to perform action %s", "RequestHandler", action.c_str());
-                    throw httpd::RequestErrorException(httpd::StatusCode::BAD_REQUEST);
+                    throw httpd::RequestErrorException(httpd::StatusCode::BAD_REQUEST,
+                        utils::format("Action '%s' is not supported", action.c_str()));
                 }
             }
         }
         catch (httpd::RequestErrorException e)
         {
-            httpd::Response response(e.getStatusCode());
+            auto response = makeErrorResponse(e.getStatusCode(), e.getMessage());
             requestLogger.setResponse(response);
+            requestLogger.setErrorMessage(e.getMessage());
             return response;
         }
-        catch (nlohmann::detail::parse_error)
+        catch (nlohmann::detail::parse_error e)
         {
             logger::warn("Error parsing json", "RequestHandler");
-            httpd::Response response(httpd::StatusCode::BAD_REQUEST);
+            std::string errorMessage = "Invalid json format";
+            auto response = makeErrorResponse(httpd::StatusCode::BAD_REQUEST, errorMessage);
             requestLogger.setResponse(response);
+            requestLogger.setErrorMessage(errorMessage);
             return response;
         }
-        catch (std::exception)
+        catch (nlohmann::detail::exception e)
+        {
+            logger::warn("Error processing json", "RequestHandler");
+            auto response = makeErrorResponse(httpd::StatusCode::BAD_REQUEST, e.what());
+            requestLogger.setResponse(response);
+            requestLogger.setErrorMessage(e.what());
```

Comment:
Here we catch all nlohmann exception other than `parse_error`. That includes the `type_error`. `what` will be printed on logs and sent as message on response. @olofkallander, do you think this is enough?